### ISSUE TITLE
`SLF001`: False negative when `self`/`cls` not direct ancestor

### ISSUE THREAD

**Issue by ngnpope** [charliermarsh/ruff#2521]
Here's some example code:
```python
class Request:
    _private = True

class View:
    def __init__(self, request):
        self.request = request

    def process():
        if self.request._private:  # False Negative
            return None
        return self.request

request = Request()
view = View(request)
assert request._private  # SLF001
assert view.request._private  # False Negative
```
Which yields the following:
```console
❯ ruff --version
ruff 0.0.240

❯ ruff --select SLF001 bug.py 
bug.py:15:8: SLF001 Private member accessed: `request._private`
Found 1 error.
```

@saadmk11 Linking you in to this one... Thanks for [implementing](https://github.com/charliermarsh/ruff/pull/2470) this plugin 🦸🏻

**Comment by saadmk11** [comment]
Oops! Missed a case. Thanks for finding it. :)
